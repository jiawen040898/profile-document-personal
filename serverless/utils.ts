export const importValue = (exportName: string): SafeAny =>
    // eslint-disable-next-line @typescript-eslint/naming-convention
    ({ 'Fn::ImportValue': exportName });
