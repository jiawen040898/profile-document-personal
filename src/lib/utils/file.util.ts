import { FileContentType, mapFileExtensionToContentType } from '@pulsifi/enum';
import { File } from '@pulsifi/interfaces';

const getMimeType = (file: File): string => {
    /*
     * Determine the MIME type of file:
     * - If the file's MIME type is 'application/octet-stream' or undefined, it means the file type is unknown.
     * - In this case, we try to determine the MIME type from the file name.
     * - If the file's MIME type is known, we simply return it.
     */

    return file.mimeType &&
        file.mimeType !== FileContentType.APPLICATION_OCTET_STREAM
        ? file.mimeType
        : getMimeTypeFromFileName(file.filename);
};

const getFileExtension = (fileName: string): string => {
    return fileName.split('.').pop() ?? '';
};

const getMimeTypeFromFileName = (fileName: string): string => {
    const fileExtension = `.${getFileExtension(fileName)}`;

    return (
        mapFileExtensionToContentType[fileExtension] ??
        FileContentType.APPLICATION_OCTET_STREAM
    );
};

export const FileUtil = {
    getMimeType,
    getFileExtension,
    getMimeTypeFromFileName,
};
