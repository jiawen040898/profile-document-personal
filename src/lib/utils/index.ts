export * from './file.util';
export * from './s3-tag.util';
export * from './strings.util';
