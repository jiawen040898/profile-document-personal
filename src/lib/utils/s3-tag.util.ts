import { Tag } from '@aws-sdk/client-s3';

export const s3TagUtil = {
    buildTagString: (tag: { [key: string]: string }): string => {
        const tagStrings = Object.entries(tag)
            .map(([key, value]) => `${key}=${value}`)
            .join('&');

        return tagStrings;
    },

    buildTagSet: (tags: { [key: string]: string }): Tag[] => {
        const tagSet: Tag[] = [];

        for (const key of Object.keys(tags)) {
            tagSet.push({ Key: key, Value: tags[key] });
        }

        return tagSet;
    },

    findTagKey: (tagSet: Tag[], key: string): Tag | undefined => {
        return tagSet.find((i) => i.Key === key);
    },

    getTagKeyValue: (tagSet: Tag[], key: string): string => {
        const keyFound = tagSet.find((i) => i.Key === key);

        return keyFound?.Value ?? '';
    },
};
