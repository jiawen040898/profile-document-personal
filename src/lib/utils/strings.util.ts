export function sanitizeByteString(input?: string | null): string | null {
    return input?.replace(/\x00/g, '') || null;
}
