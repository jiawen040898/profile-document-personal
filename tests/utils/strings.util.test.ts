import { sanitizeByteString } from '@pulsifi/utils';

describe('sanitizeByteString', () => {
    it('removes null bytes from input string', () => {
        // Arrange
        const input =
            '● I have 10 years of exp\u0000erience in personal finance management, and I have assisted 45 rep\u0000eat clients in increasing their capital by an ave\u0000rage of 15% every year.\n● I have 10 years of experience in personal finance management, and I have assisted 45 repeat clients in increasing their ca\u0000pital by an average of 15% every year.\n';

        // Assert
        expect(sanitizeByteString(input)).toMatchSnapshot();
    });

    it('returns null if input is null', () => {
        // Assert
        expect(sanitizeByteString(null)).toBe(null);
    });

    it('returns null if input is undefined', () => {
        // Assert
        expect(sanitizeByteString(undefined)).toBe(null);
    });

    it('returns empty string if input is an empty string', () => {
        // Assert
        expect(sanitizeByteString('')).toBe(null);
    });

    it('returns input string if there are no null bytes', () => {
        // Assert
        expect(sanitizeByteString('hello')).toBe('hello');
    });
});
