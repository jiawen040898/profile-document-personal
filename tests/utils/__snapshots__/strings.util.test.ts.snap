// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sanitizeByteString removes null bytes from input string 1`] = `
"● I have 10 years of experience in personal finance management, and I have assisted 45 repeat clients in increasing their capital by an average of 15% every year.
● I have 10 years of experience in personal finance management, and I have assisted 45 repeat clients in increasing their capital by an average of 15% every year.
"
`;
